Apps/phcuckoo/zip_and_encrypt.py

```diff
@@ -0,0 +1,102 @@
+#! /usr/bin/env python3
+
+zipexec = "/usr/bin/zip"
+
+import sys, os
+import uuid
+import subprocess
+
+def _is_exec(path):
+    if not os.path.isfile(path) or not os.access(path, os. X_OK):
+        return False
+    return True
+
+class zip_and_encrypt:
+    def __init__(self, prefix, password):
+        if not _is_exec(zipexec):
+            raise Exception(f"{path} is not executable")
+
+        self._prefix = prefix
+        self._password = password
+        self._archive = f"{prefix}_{uuid.uuid4()}"
+        self._fp = None
+        self._output = None
+        
+    @property
+    def output(self):
+        return self._output
+
+    @property
+    def archive_name(self):
+        return self._archive
+
+    @property
+    def archive_path(self):
+        return self._archive + ".zip"
+
+    @property
+    def archive_fp(self):
+        if self._fp:
+            return self._fp
+
+        if not os.path.isfile(self.archive_path):
+            raise Exception(f"{self.archive_path} doesn't exists")
+
+        self._fp = open(self.archive_path, "rb")
+        return self._fp
+
+    def add(self, filename):
+        if not os.path.exists(filename):
+            raise Exception(f"{filename} does not exists")
+
+        cmd = [
+            zipexec,
+            "-j",
+            "-P",
+            self._password,
```

Comment:
can we avoid passing a password in the command and pass it through stdin instead?

looking at zip's man page I think using the `-e` flag should achieve this. although you may need to use something like `pexpect` to handle the interaction if direct piping doesn't work